h5netcdf/core.py

```diff
@@ -552,9 +561,20 @@ def resize_dimension(self, dimension, size):
 class File(Group):
 
     def __init__(self, path, mode='a', invalid_netcdf=None, **kwargs):
-        self._preexisting_file = os.path.exists(path)
         try:
-            self._h5file = h5py.File(path, mode, **kwargs)
+            if path.startswith(('http', 'hdf5:')):
+                if no_h5pyd:
+                    raise RuntimeError('h5pyd package required')
+                try:
+                    with h5pyd.File(path, 'r') as f:  # noqa
+                        pass
+                    self._preexisting_file = True
+                except (OSError, IOError):
```

Comment:
Can h5pyd really raise both these errors if a file isn't found? I would think one would be enough...

Reply:
This issue came up when testing in Travis with Python 2.7. For Pythons 3: `OSError`; for Python 2.7: `IOError`.

Reply:
OK, please add a brief comment here to explain.

Looking at the [Python docs](https://docs.python.org/3/library/exceptions.html#IOError):

> The following exceptions are kept for compatibility with previous versions; starting from Python 3.3, they are aliases of OSError.
>
> exception EnvironmentError
> exception IOError
> exception WindowsError
> Only available on Windows.

So I think it would probably be OK to use `IOError` for both cases.